terraform/gcp/cluster.tf

```diff
@@ -32,6 +12,26 @@ resource "google_container_cluster" "juniper_cluster" {
     stack_type                    = "IPV4_IPV6"
   }
 
+  cluster_autoscaling {
```

Comment:
how does this autoscaling property interface with autopilot?

Reply:
The terraform docs are really bad here. I don't think it makes any sense for the config for this to be here, but, in fact, this is the _correct_ way of doing it with autopilot. Normally, you'd use `node_config` to change the service account, but unfortunately, that does not work with autopilot, so you have to use this arcane nested structure to change the service account.

You can see it's still autopilot:
![image](https://github.com/user-attachments/assets/5455da31-0b1f-4fd0-b24c-032f041eaeb0)
And it's using a custom service account:
![image](https://github.com/user-attachments/assets/105a88a3-b730-4604-a495-93d8b7a2800d)